### PR TITLE
Fixing cookie header parsing logic

### PR DESCRIPTION
**Pull request by nickmelis**
Addressing cookie header parsing issues discussed in https://github.com/slackapi/java-slack-sdk/issues/555#issuecomment-693398632

The existing parsing logic may not work in presence of more than one cookie.
I added a new test to cover the scenario I encountered in production, together with a change in an existing test to cover the scenario where two cookies are received as list of strings.

@seratch as discussed before. Let me know what you think and if any change is needed

### Category (place an `x` in each of the `[ ]`)

* [x] **bolt** (Bolt for Java)
* [ ] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

### CODE REVIEW

File: bolt/src/test/java/test_locally/service/OAuthStateServiceTest.java [hunk 1]
```diff
@@ -98,6 +108,21 @@ public void isValid_different_param_value() {
...
+      query.put("state", Arrays.asList("123", "234"));
+      Map<String, List<String>> rawHeaders = new HashMap<>();
+      rawHeaders.put("Cookie", Arrays.asList(
+        "__cookie1=abc; _cookie2=def; " + service.getSessionCookieName() + "=123"));
+      RequestHeaders headers = new RequestHeaders(rawHeaders);
+      VerificationCodePayload payload = new VerificationCodePayload();
+      payload.setState("123");
+      boolean valid = service.isValid(new OAuthCallbackRequest(query, "", payload, headers));
+      assertTrue(valid);
+    }
```
Comment (seratch) [inline bolt/src/test/java/test_locally/service/OAuthStateServiceTest.java:124]:
```suggestion
    @Test
    public void isValid_multiple_cookies() {
        Map<String, List<String>> query = new HashMap<>();
        query.put("foo", Arrays.asList("bar", "baz"));
        query.put("state", Arrays.asList("123", "234"));
        Map<String, List<String>> rawHeaders = new HashMap<>();
        rawHeaders.put("Cookie", Arrays.asList(
                "__cookie1=abc; _cookie2=def; " + service.getSessionCookieName() + "=123"));
        RequestHeaders headers = new RequestHeaders(rawHeaders);
        VerificationCodePayload payload = new VerificationCodePayload();
        payload.setState("123");
        boolean valid = service.isValid(new OAuthCallbackRequest(query, "", payload, headers));
        assertTrue(valid);
    }
```

**Review by seratch — CHANGES_REQUESTED** [review]
Could you accept my suggestions for consistent indents? Apart from these changes, LGTM

**Comment by seratch** [comment]
@nickmelis Thanks for fixing this! Except for the indents, this PR already looks great to me. I will backport this to 1.1.x series and will release a new patch version very soon.

**Comment by seratch** [comment]
@nickmelis Also, would you mind signing the CLA (check the above comment by a bot user)? Without it, we are unable to merge your changes.

**Comment by nickmelis** [comment]
> @nickmelis Also, would you mind signing the CLA (check the above comment by a bot user)? Without it, we are unable to merge your changes.

@seratch I did sign it, for some reason it's still showing as not signed. Any idea why?

I accepted all your suggestions, so formatting should now be sorted. Is there anything left for me to do?

**Comment by seratch** [comment]
@nickmelis You need to use the email address that is associated with your GitHub account in commits. You don't need to keep the commits by me. Could you squash the commits into one commit using a valid email address?

**Comment by nickmelis** [comment]
> @nickmelis You need to use the email address that is associated with your GitHub account in commits. You don't need to keep the commits by me. Could you squash the commits into one commit using a valid email address?

Ok, _I think_ I squashed the 3 commits into one using my nick@cloudfle.com email that is set here on GitHub. But CLA is still showing as "Not signed yet". What is missing?

**Comment by seratch** [comment]
@nickmelis As you can see you still have 5 commits in this branch. https://github.com/slackapi/java-slack-sdk/pull/560/commits

Can you try the following steps?

```
git reset HEAD~5
git add .
git commit -m'Fix cookie header parsing logic'
git push origin main -f
```

**Comment by nickmelis** [comment]
Ok @seratch, that seems to have worked! Thanks for your help!

**Review by seratch — APPROVED** [review]

**Pull request merged**